[14.0] [MIG] account_(bank_)statement_import_online(_ponto)

NL66278 opened the pull request:
Migrate (and rename) both account_bank_statement_import_online as well as account_(bank_)statement_import_online_ponto.

The ... _ponto is the only online provider implemented in 13.0, and obviously it can not be migrated without the base module.

alexey-pelykh committed:
[ADD] account_bank_statement_import_online: online bank statements

[UPD] Update account_bank_statement_import_online.pot

[UPD] README.rst

[ADD] icon.png

alexey-pelykh committed:
[FIX] account_bank_statement_import_online: catch any exception

[UPD] Update account_bank_statement_import_online.pot

account_bank_statement_import_online 12.0.1.0.1

alexey-pelykh committed:
[IMP] account_bank_statement_import_online: hide Create/Import statement

account_bank_statement_import_online 12.0.1.1.0

alexey-pelykh committed:
[IMP] account_bank_statement_import_online: use pre-computed sanitized bank account number

[UPD] Update account_bank_statement_import_online.pot

account_bank_statement_import_online 12.0.1.2.0

alexey-pelykh committed:
[IMP] account_bank_statement_import_online: allow pull if archived

[UPD] Update account_bank_statement_import_online.pot

account_bank_statement_import_online 12.0.1.3.0

alexey-pelykh committed:
[FIX] account_bank_statement_import_online: correct balance recalculation

account_bank_statement_import_online 12.0.1.3.1

fmdl committed:
[ADD] account_bank_statement_import_online_ponto

alexey-pelykh committed:
[IMP] account_bank_statement_import_online: tz

[UPD] Update account_bank_statement_import_online.pot

account_bank_statement_import_online 12.0.1.4.0

alexey-pelykh committed:
[IMP] account_bank_statement_import_online: add maintainer

[UPD] README.rst

alexey-pelykh committed:
[IMP] account_bank_statement_import_online: better logging

[UPD] Update account_bank_statement_import_online.pot

account_bank_statement_import_online 12.0.1.4.1

alexey-pelykh committed:
[IMP] account_bank_statement_import_online: black, isort, prettier

alexey-pelykh committed:
[MIG] account_bank_statement_import_online: Migration to 13.0

alexey-pelykh committed:
[IMP] account_bank_statement_import_online: always create a statement

alexey-pelykh committed:
[FIX] account_bank_statement_import_online: tests

alexey-pelykh committed:
[FIX] account_bank_statement_import_online: correct range filtering

pedrobaeza committed:
[FIX+IMP] account_bank_statement_import_online_ponto: README by fragments + avoid tz conversion + icon

oca-travis committed:
[UPD] Update account_bank_statement_import_online.pot

OCA-git-bot committed:
[UPD] README.rst

alexey-pelykh committed:
[UPD] Brainbean Apps => CorporateHub

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
[FIX+IMP] account_bank_statement_import_online_ponto: Adjustments for better use

* FIX: Returned dates are in UTC, so they were classified incorrectly in
  the bad statement.
* FIX: Information in ref was the most important one, being the other
  mostly '/'
* IMP: Preserve all possible information from the source, including
  counterpart information.
* IMP: Fill Odoo statement line fields for account number and partner
  name for better support.

pedrobaeza committed:
[FIX] account_bank_statement_import_online_ponto: flake8

joao-p-marques committed:
[IMP] account_bank_statement_import_online_ponto: black, isort, prettier

joao-p-marques committed:
[MIG] account_bank_statement_import_online_ponto: Migration to 13.0

TT26550

oca-travis committed:
[UPD] Update account_bank_statement_import_online_ponto.pot

bosd committed:
Added translation using Weblate (Dutch)

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 92.0% (81 of 88 strings)

Translation: bank-statement-import-13.0/bank-statement-import-13.0-account_bank_statement_import_online
Translate-URL: https://translation.odoo-community.org/projects/bank-statement-import-13-0/bank-statement-import-13-0-account_bank_statement_import_online/nl/

NL66278 committed:
[IMP] account_bank_statement_import_online: black, isort, prettier

NL66278 committed:
[IMP] account_bank_statement_import_online_ponto: black, isort, prettier

NL66278 committed:
[IMP] Adapt account_bank_statement_import_online to 14.0

Also Rename account_bank_statement_import_online => account_statement_import_online

alexis-via commented:
Requires PR https://github.com/OCA/web/pull/1934

alexis-via commented:
I can't even install account_statement_import_online of this PR on a v14 database:
- data/account_bank_statement_import_online.xml : the module name in model_id must be renamed
- views/account_journal.xml: there is an inherit of account.view_account_bank_journal_form which doesn't exist in v14. 
- in the same file, there is an inherit of account_bank_statement_import.journal_dashboard_view_inherit => module name is wrong in v14
@NL66278 What happened ? Maybe you pushed the wrong code?

NL66278 commented:
@alexis-via My apologies, should have made clear that this was still WIP. Thanx for pointing out the missing parts anyway!

alexis-via commented:
There is still several things to update:
- fix a view that prevent the module from being installed
- in online_bank_statement_provider.py: use of journal.sequence_id => sequence_id field doesn't exist any more in v14

NL66278 commented:
@alexis-via Should be good now. Hope to have a customer account for real live testing in a few days.

alexis-via committed:
account_statement_import_online: usability improvement on form view of online.bank.statement.provider: avoid speading the information on 3 tabs. As there are just a few fields, let's keep all the information directly visible.

Fix warning on search view of online.bank.statement.provider
Fix some tests
Button on journal form view to access the wizard
The wizard now returns the bank statements of the journal(s)
Fix wired strings

alexis-via commented:
I improved the PR a bit in this PR: https://github.com/NL66278/bank-statement-import/pull/8

NL66278 committed:
[FIX] account_statement_import_online: further improvements.

ernestotejeda commented:
@NL66278 , since I need this module and there is also a mix of migration commits, I will proceed to supersede this PR. Is it OK for you?

NL66278 commented:
@ernestotejeda Please elaborate? I am fixing up a last test that is failing (should not effect actual function). If you have any extra change, you could submit it as a PR to my branch?

ernestotejeda commented:
@NL66278  If you are working on the PR, it's OK :). Anyway I suggest, please, that you leave only two commits for each module, one for pre-commit related changes and another one for migration related changes.

ernestotejeda reviewed (commented):
Code review: :+1:

ernestotejeda reviewed (commented):
@NL66278 I don't see the pre-commit commit.

pedrobaeza commented:
@NL66278 Please avoid merge commits in any PR, and squash your migrations commits according guidelines.

ernestotejeda commented on account_statement_import_online/README.rst:
```diff
@@ -0,0 +1,127 @@
+
+To configure online bank statements provider:
+
+#. Go to *Invoicing > Configuration > Bank Accounts*
```
This menu no longer exist.

ernestotejeda reviewed (commented):
Code review: :+1: 
I approve the changes made to the code, but I would have liked more that each commit had changes related to a single module. Anyway LGTM.

pedrobaeza commented:
/ocabot merge nobump
